src/arena.c

```diff
@@ -372,9 +384,7 @@ arena_extent_ralloc_large_shrink(tsdn_t *tsdn, arena_t *arena, edata_t *edata,
 	size_t usize = edata_usize_get(edata);
 
 	if (config_stats) {
-		LOCKEDINT_MTX_LOCK(tsdn, arena->stats.mtx);
```

Comment:
One effect of moving the locking downstream, is that the ralloc now requires two lock-unlock pairs instead of one (I think this was why the locks were placed outside). Since this only applies to the realloc + no-64bit-atomics case, I don't think it's really a problem.